dev_scripts_helpers/github/sync_gh_issue_labels.py

```diff
@@ -0,0 +1,326 @@
+                    description=item["description"],
+                    color=item["color"],
+                )
+                for item in yaml_data
+            ]
+            return labels
+    except Exception as e:
+        _LOG.error("Error loading label inventory manifest file: %s", str(e))
+        sys.exit(1)
+
+
+def save_labels(labels: List[Label], path: str) -> None:
+    """
+    Save labels to the label inventory manifest file.
+
+    :param labels: label objects
+    :param path: path to save the label inventory manifest file to
+    """
+    try:
+        with open(path, "w") as file:
+            yaml_data = [
+                {
+                    "name": label.name,
+                    "description": label.description,
+                    "color": f"#{label.color}",
+                }
+                for label in labels
+            ]
+            yaml.dump(yaml_data, file, default_flow_style=False, sort_keys=False)
+    except Exception as e:
+        _LOG.error("Error saving label inventory manifest file: %s", str(e))
+        sys.exit(1)
+
+
+def sync_labels(
+    client: GitHubClient, owner: str, repo: str, labels: List[Label], prune: bool
+) -> None:
+    """
+    Synchronize labels between config and repository.
+
+    :param client: GitHub client
+    :param owner: GitHub repository owner/organization
+    :param repo: GitHub repository name
+    :param labels: label objects
+    :param prune: delete labels that exist in the repo but not in the
+        label inventory manifest file
+    """
+    # Build maps for efficient lookup.
+    label_map = {label.name: label for label in labels}
+    current_labels = client.get_labels(owner, repo)
+    current_label_map = {label.name: label for label in current_labels}
+    # Delete labels if pruning is enabled.
+    if prune:
+        for current_label in current_labels:
+            if current_label.name not in label_map:
+                try:
+                    client.delete_label(owner, repo, current_label.name)
+                except Exception as e:
+                    _LOG.error(
+                        "Error deleting label %s: %s", current_label.name, e
+                    )
+    # Create or update labels.
+    for label in labels:
+        current_label = current_label_map.get(label.name)
+        try:
+            if current_label is None:
+                # Label doesn't exist, create it.
+                client.create_label(owner, repo, label)
+            elif (
+                current_label.description != label.description
+                or current_label.color != label.color
+            ):
+                # Label exists but needs update.
+                client.update_label(owner, repo, label)
+            else:
+                # Label exists and is identical.
+                _LOG.info("Label not changed: %s", label)
+        except Exception as e:
+            _LOG.error("Error processing label %s: %s", label.name, e)
```

Comment:
Same thing here. This is actually making the error worse, since it's better to get info about the failing point.

Long story short, we use try-except very rarely since it just makes the code more complicated.